backlog_controller.hh

```diff
@@ -130,13 +130,16 @@ public:
     static constexpr float backlog_disabled(float backlog) { return std::isinf(backlog); }
     static inline const std::vector<backlog_controller::control_point> default_control_points = {
             backlog_controller::control_point{0.0, 50}, {1.5, 100}, {normalization_factor, 1000}};
-    compaction_controller(backlog_controller::scheduling_group sg, float static_shares, std::chrono::milliseconds interval, std::function<float()> current_backlog)
+    compaction_controller(backlog_controller::scheduling_group sg, float static_shares, float max_shares,
```

Comment:
It's nicer to use std::optional (or +inf) for "no maximum" rather than a magic value that is in fact lower than the actual shares.

Reply:
Agreed. I am thinking whether we should reject a value of 0, and enforce a minimum (e.g. output of lowest control point = 50). Nobody with good sanity would set max shares to lower than 50. What does it happen when we set shares to 0? very slow progress or none? I guess the former. It might even mess with the scheduling calculation, perhaps. You set shares to quite low value, and quite soon you might have all your tables suffering with terrible read amplification. We don't want this to happen even by accident so better to adjust to some minimum sane value.